<<<<<<< HEAD:Waterfall-Proxy-Patches/0014-Don-t-allow-name-containing-dot.patch
From 1d00df675017b9bedd81cbf07174e727dd7d1056 Mon Sep 17 00:00:00 2001
=======
From 9754f77c4142bebe9425d493b6826503dd9060f9 Mon Sep 17 00:00:00 2001
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0013-Don-t-allow-name-containing-dot.patch
From: foss-mc <69294560+foss-mc@users.noreply.github.com>
Date: Wed, 16 Dec 2020 18:25:51 +0800
Subject: [PATCH] Don't allow name containing dot


diff --git a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
index 1811084a..1925188f 100644
--- a/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
+++ b/proxy/src/main/java/net/md_5/bungee/connection/InitialHandler.java
@@ -387,7 +387,10 @@ public class InitialHandler extends PacketHandler implements PendingConnection
         Preconditions.checkState( thisState == State.USERNAME, "Not expecting USERNAME" );
         this.loginRequest = loginRequest;
 
-        if ( getName().contains( " " ) )
+        // FlameCord - Use name
+        final String name = getName();
+
+        if ( name.contains( " " ) || name.contains( "." ) )
         {
             disconnect( bungee.getTranslation( "name_invalid" ) );
             return;
-- 
2.32.0.windows.1

